src/m365/spo/commands/app/SpoAppBaseCommand.ts

```diff
@@ -13,36 +13,39 @@ interface Options extends GlobalOptions {
 }
 
 export abstract class SpoAppBaseCommand extends SpoCommand {
-  public getAppCatalogSiteUrl(logger: Logger, authSiteUrl: string, args: CommandArgs): Promise<string> {
-    return new Promise<string>((resolve: (appCatalogSiteUrl: string) => void, reject: (error: any) => void): void => {
-      if (args.options.appCatalogScope === 'sitecollection') {
-        return resolve((args.options.appCatalogUrl as string).toLowerCase().replace('/appcatalog', ''));
-      }
+  protected async getAppCatalogSiteUrl(logger: Logger, authSiteUrl: string, args: CommandArgs): Promise<string> {
+    if (args.options.appCatalogScope === 'sitecollection') {
+      // Trim any trailing slashes if there are any
+      const appCatalogUrl = args.options.appCatalogUrl!.replace(/\/$/, '');
+      const appCatalogUrlChunks = appCatalogUrl.split('/');
 
-      if (args.options.appCatalogUrl) {
-        return resolve(args.options.appCatalogUrl);
+      // Trim the last part of the URL if it ends on '/appcatalog', but don't trim it if the site URL is called like that (/sites/appcatalog).
+      if (appCatalogUrl.toLowerCase().endsWith('/appcatalog') && appCatalogUrlChunks.length !== 5) {
```

Comment:
Rather than looking at the number of chunks, wouldn't it be more reliable to look specifically for `/sites/appcatalog`?

Reply:
Looks like it's even more complex than I thought. Just found a tenant where I can set up `/sites/section_fg_ekek_7a/` as managed path.

![afbeelding](https://github.com/pnp/cli-microsoft365/assets/11723921/983d0761-a241-40e0-9097-cd68c3ab0186)

I actually don't know how you can add this. All online documentation says it's impossible to add new managed paths. 😄
This managed path (`/sites/section_fg_ekek_7a/`) kinda breaks my code tho. I was aware you could choose between `sites` and `teams` but now I know this, I will have to update my code.

Reply:
Thinking of it some more, this makes it totally unpredictable for us to see which part is the site URL.
Example: the URL `https://contoso.sharepoint.com/sites/section_fg_ekek_7a/AppCatalog`

This could be:

- AppCatalog URL of the site `https://contoso.sharepoint.com/sites/section_fg_ekek_7a`
- An absolute URL pointing to a site `https://contoso.sharepoint.com/sites/section_fg_ekek_7a/AppCatalog`

This makes it kinda impossible for us to check if a site URL or app catalog URL was specified. Since normally most tenants can only choose between `/sites` and `/teams` managed path and I can't really find a way to create your own managed path, maybe we should just stick with the default behavior and ignore edge cases with another managed path like `/sites/section_fg_ekek_7a` in this case.

Reply:
> maybe we should just stick with the default behavior and ignore edge cases with another managed path like /sites/section_fg_ekek_7a in this case.

Makes sense